src/MongoDB.Driver/FilteredMongoCollectionBase.cs

```diff
@@ -157,6 +182,16 @@ public override Task<IAsyncCursor<TField>> DistinctAsync<TField>(IClientSessionH
             return _wrappedCollection.DistinctAsync(session, field, CombineFilters(filter), options, cancellationToken);
         }
 
+        public override long EstimatedDocumentCount(EstimatedDocumentCountOptions options = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            throw new NotSupportedException("EstimatedDocumentCount is not supported for filtered collections.");
```

Comment:
What is FilteredMongoCollection used for?

Reply:
It is part of the implementation of the OfType<TSubclass> method of IMongoCollection which returns an IFilteredMongoCollection.